pkg/lua/contrib/binsearch.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @file
+ *
+ * @brief   Generic binary search for tables containing strings.
+ * @author  Juan Carrano <j.carrano@fu-berlin.de>
+ *
+ */
+
+#include <string.h>
+
+#include "binsearch.h"
+
+int binsearch_str(const void *start, size_t offset, size_t stride, size_t nmemb,
+                  const char *str, size_t n)
+{
+    const uint8_t *cstart = (((const uint8_t *)start) + offset);
+    size_t lo = 0, hi = nmemb;
+
+    while (lo < hi) {
+        size_t mid = (lo + hi) / 2;
+        const char *target = *((const char * const *)(cstart + mid * stride));
+        int cmp = strncmp(str, target, n);
+
+        if (cmp == 0)
+            return mid;
+        else if (cmp < 0)
+            hi = mid;
+        else /* (cmp > 0) */
+            lo = mid + 1;
+    }
+    return BINSEARCH_NOTFOUND;
```

Comment:
if possible, prefer `errno` types

Reply:
I searched, but I cannot find an errno code that matches (they concern mostly external errors). Anyways, there is only one error that can be thrown by a search routine: "not found".

Reply:
ENOTFOUND should do the trick :)

Reply:
`ENOTFOUND` is not in `errno.h` and is not defined in RIOT either.

Reply:
you can use ENOENT as seen [here](https://github.com/RIOT-OS/RIOT/blob/af2c1538898db82006e006e43c3af5bbd099e8b4/sys/net/gnrc/netif/hdr/gnrc_netif_hdr.c#L65)

Reply:
Mmmhh, I disagree with that usage. [The definition](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/errno.h.html) says it means "No such file or directory". I'll change anyways, it if is a sort of convention in RIOT to use it that way.

Reply:
ENOENT stands for Error No ENTry (or Error No ENTity), as seen [here](https://stackoverflow.com/questions/19902828/why-does-enoent-mean-no-such-file-or-directory). It's usually described as "No such file or directory", but can be used for more cases